sequencer/src/api.rs

```diff
@@ -434,24 +430,54 @@ pub mod test_helpers {
         P: PersistenceOptions,
         C: StateCatchup + 'static,
     {
+        /// NOTICE: This function will reset most of the config. Please assign the number of the nodes
+        /// before assigning values.
```

Comment:
Perhaps this is better as a constructor then, i.e. `TestNetworkConfigBuilder::with_num_nodes::<3>()` rather than say `TestNetworkConfigBuilder::default().num_nodes(3)`. Then we could put all the default values here and implement default as `Self::with_num_nodes::<5>()`

Reply:
https://github.com/EspressoSystems/espresso-sequencer/pull/1641/commits/819528ef6c0d29606805fbe2401966e85695d54d